Test pr ci

amore17 opened the pull request.

a-sharov committed:
Add ipa-idrange-fix

ipa-idrange-fix is a tool for analysis of existing IPA ranges, users
and groups outside of those ranges, and functionality to propose
and apply remediations to make sure as much users and groups as
possible end up in the IPA-managed ranges.

Fixes: https://pagure.io/freeipa/issue/9612

Signed-off-by: Aleksandr Sharov <asharov@redhat.com>
Reviewed-By: Rob Crittenden <rcritten@redhat.com>

a-sharov committed:
Add PR-CI definitions

fixes: https://pagure.io/freeipa/issue/9612
Reviewed-By: Rob Crittenden <rcritten@redhat.com>

mreynolds389 committed:
ipa-migrate - fix migration issues with entries using ipaUniqueId in the RDN

We need to handle these entries differently and specify what attribute
and search base to use to find the entry on the local server.  Most
entries can use the "cn" attribute but for selinux usermaps we need to
search using the ipaOwner attribute which is a DN, and in turn requires
additional handling/converting in order to properly check if the usermap
exists or not.

Also fixed an issue where an attribute should be removed from the local
entry if it does not exist on the remote entry.

And fixed the handling od "sudoOrder" which is defined as multi-valued
in the schema, but we really need to treat it as single-valued

Fixes: https://pagure.io/freeipa/issue/9640

Signed-off-by: Mark Reynolds <mreynolds@redhat.com>
Reviewed-By: Rob Crittenden <rcritten@redhat.com>

f-trivino committed:
Installer: activate ssh service in sssd.conf

This commit enables SSSD's ssh service in ipa-client-install to ensure
sss_ssh_knownhosts and sss_ssh_knownhostsproxy functions properly.

Fixes: https://pagure.io/freeipa/issue/9649
Related: https://pagure.io/freeipa/issue/9536

Signed-off-by: Francisco Trivino <ftrivino@redhat.com>
Reviewed-By: Rob Crittenden <rcritten@redhat.com>

mrizwan93 committed:
ipatests: refactor password file handling in TestHSMInstall

When token and associated certs are not being cleaned
up properly, the subsequent installation fails. Hence
Password file related scenarios moved out to new test class
so that it have fresh installation.

Signed-off-by: Mohammad Rizwan <myusuf@redhat.com>
Reviewed-By: Rob Crittenden <rcritten@redhat.com>
Reviewed-By: Florence Blanc-Renaud <frenaud@redhat.com>

amore17 committed:
ipatests: Update ipatests to test topology with multiple domain.

Added changes in ipatests so that ipa server-replica-client
can be installed with two domain - ipa.test and trustedipa.test

Related: https://pagure.io/freeipa/issue/9657

Signed-off-by: Anuja More <amore@redhat.com>

abbra committed:
selinux: add all IPA log files to ipa_log_t file context

We have multiple log files that produced by IPA components. Some of them
are written by the tools that run as root and inherit their file context
from /var/log -> var_log_t. However, increasingly we get tools that were
run through oddjob helpers. These supposed to be run within ipa_helper_t
SELinux context which has write permissions for ipa_log_t file context.

Add all known log files from the base platform. The following script was
used to generate them:
$ git grep '_LOG = .*ipa.*\.log' ipaplatform/base/paths.py | cut -d= -f2 | \
      xargs -I% echo -e "%\t--\tgen_context(system_u:object_r:ipa_log_t,s0)"

/var/log/ipabackup.log	--	gen_context(system_u:object_r:ipa_log_t,s0)
/var/log/ipaclient-install.log	--	gen_context(system_u:object_r:ipa_log_t,s0)
/var/log/ipaclient-uninstall.log	--	gen_context(system_u:object_r:ipa_log_t,s0)
/var/log/ipaclientsamba-install.log	--	gen_context(system_u:object_r:ipa_log_t,s0)
/var/log/ipaclientsamba-uninstall.log	--	gen_context(system_u:object_r:ipa_log_t,s0)
/var/log/ipareplica-ca-install.log	--	gen_context(system_u:object_r:ipa_log_t,s0)
/var/log/ipareplica-conncheck.log	--	gen_context(system_u:object_r:ipa_log_t,s0)
/var/log/ipareplica-install.log	--	gen_context(system_u:object_r:ipa_log_t,s0)
/var/log/iparestore.log	--	gen_context(system_u:object_r:ipa_log_t,s0)
/var/log/ipaserver-enable-sid.log	--	gen_context(system_u:object_r:ipa_log_t,s0)
/var/log/ipaserver-install.log	--	gen_context(system_u:object_r:ipa_log_t,s0)
/var/log/ipaserver-adtrust-install.log	--	gen_context(system_u:object_r:ipa_log_t,s0)
/var/log/ipaserver-dns-install.log	--	gen_context(system_u:object_r:ipa_log_t,s0)
/var/log/ipaserver-kra-install.log	--	gen_context(system_u:object_r:ipa_log_t,s0)
/var/log/ipaserver-uninstall.log	--	gen_context(system_u:object_r:ipa_log_t,s0)
/var/log/ipaupgrade.log	--	gen_context(system_u:object_r:ipa_log_t,s0)
/var/log/ipatrust-enable-agent.log	--	gen_context(system_u:object_r:ipa_log_t,s0)
/var/log/ipaepn.log	--	gen_context(system_u:object_r:ipa_log_t,s0)
/var/log/ipa-custodia.audit.log	--	gen_context(system_u:object_r:ipa_log_t,s0)
/var/log/ipa-migrate.log	--	gen_context(system_u:object_r:ipa_log_t,s0)

ipa-custodia.audit.log was already in the present list.

Additionally, ipa-migrate-conflict.ldif is used by the ipa-migrate tool
but is not provided through the ipaplatform mechanism. It is added
explicitly.

Fixes: https://pagure.io/freeipa/issue/9654

Signed-off-by: Alexander Bokovoy <abokovoy@redhat.com>
Reviewed-By: Florence Blanc-Renaud <flo@redhat.com>

b3lix committed:
ipatests: Update ipa-adtrust-install test

update test_user_connects_smb_share_if_locked_specific_group with wait
for SSSD to be online after ipa-adtrust-install command

Related: https://pagure.io/freeipa/issue/9655

Signed-off-by: Erik Belko <ebelko@redhat.com>
Reviewed-By: Alexander Bokovoy <abokovoy@redhat.com>

flo-renaud committed:
ipatests: Add missing comma in test_idrange_no_rid_bases_reversed

The test is calling ipa idrange-add but is missing a comma in
the arguments list.
The resulting call is using "--rid-base 100300000--secondary-rid-base".
Add the missing comma to build the command with
"--rid-base 100300000 --secondary-rid-base"

Fixes: https://pagure.io/freeipa/issue/9656

Signed-off-by: Florence Blanc-Renaud <flo@redhat.com>
Reviewed-By: Alexander Bokovoy <abokovoy@redhat.com>

flo-renaud commented on ipatests/pytest_ipa/integration/__init__.py:
```diff
@@ -386,45 +386,78 @@ def mh(request, class_integration_logs):
-        config_class=Config,
-        _config=get_global_config(),
-    )
+    if cls.num_trusted_domains == 1:
```
Instead of having the new code below this specific test `if cls.num_trusted_domains == 1:`, I would rather keep the old structure and write something like
```
    for _i in range(cls.num_trusted_domains):
        domain_descriptions.append({
            'type': 'TRUSTED_IPA',
            'hosts':
            {
                'trusted_master': 1,
                'trusted_replica': cls.num_trusted_replicas,
                'trusted_client': cls.num_trusted_clients,
            },
        })
```
below `for _i in range(cls.num_ad_treedomains)` as this would allow to have both TRUSTED_IPA and AD.

mreynolds389 committed:
ipa-migrate - fix alternate entry search filter

Processing a filter like a DN can cause normalization issues that result
in an invalid filter. Make sure the filter is encapsulated with
parenthesis and we call replace_suffix_value() instead of
replace_suffix()

Fixes: https://pagure.io/freeipa/issue/9658

Signed-off-by: Mark Reynolds <mreynolds@redhat.com>

Fix typo in test

Reviewed-By: Florence Blanc-Renaud <flo@redhat.com>

flo-renaud commented on doc/designs/prci_checker.md:
```diff
@@ -73,6 +73,7 @@ classes:
   - "RunPytest3"
   - "RunWebuiTests"
   - "RunADTests"
+  - "RunMultiDomainPytest"
```
You can remove this line

flo-renaud commented on doc/designs/prci_checker.md:
```diff
@@ -182,7 +183,7 @@ $ python3 ipatests/prci_definitions/prci_checker.py -f ipatests/prci_definitions
 [File] gating [Jobs] 24
 ERROR: In job 'fedora-latest/test_installation_TestInstallMaster':
-  'class' field should be defined with one of the supported: ['RunPytest', 'RunPytest2', 'RunPytest3', 'RunWebuiTests', 'RunADTests']
+  'class' field should be defined with one of the supported: ['RunPytest', 'RunPytest2', 'RunPytest3', 'RunWebuiTests', 'RunADTests', 'RunMultiDomainPytest']
```
No need for this change.

flo-renaud commented on ipatests/prci_definitions/prci_jobs_spec.yaml:
```diff
@@ -9,6 +9,7 @@ classes:
   - "RunPytest3"
   - "RunWebuiTests"
   - "RunADTests"
+  - "RunMultiDomainPytest"
```
You can remove this line.

flo-renaud commented on ipatests/pytest_ipa/integration/__init__.py:
```diff
@@ -386,45 +386,78 @@ def mh(request, class_integration_logs):
+            config_class=Config,
+            _config=get_global_config(),
+        )
+    if cls.num_trusted_domains == 1:
```
Instead of having 2 separate code branches you can keep a single one and add
```
    for domain in mh.config.domains:
         if domain.type == 'IPA':
              mh.domain = domain
         elif domain.type == 'TRUSTED_IPA':
              mh.trusted_domain = domain

    [mh.master] = mh.domain.hosts_by_role('master')
    mh.replicas = mh.domain.hosts_by_role('replica')
    mh.clients = mh.domain.hosts_by_role('client')
    if mh.config.trusted_domains:
        [mh.trusted_master] = mh.trusted_domain.hosts_by_role('master')
        mh.trusted_replicas = mh.trusted_domain.hosts_by_role('replica')
        mh.trusted_clients = mh.trusted_domain.hosts_by_role('client')

```

flo-renaud commented on ipatests/pytest_ipa/integration/__init__.py:
```diff
@@ -477,15 +510,26 @@ def add_compat_attrs(cls, mh):
-        cls.ads = mh.ads
-        cls.ad_subdomains = mh.ad_subdomains
-        cls.ad_treedomains = mh.ad_treedomains
+    if cls.num_trusted_domains == 1:
```
Instead of having 2 separate code branches, you can keep the old structure and add
```
    cls.trusted_domains = mh.config.trusted_domains
    if cls.trusted_domains:
        cls.trusted_master = mh.trusted_master
        cls.trusted_replicas = mh.trusted_replicas
        cls.trusted_clients = mh.trusted_clients
```

flo-renaud commented on ipatests/pytest_ipa/integration/__init__.py:
```diff
@@ -494,15 +538,25 @@ def del_compat_attrs(cls):
-        del cls.ad_subdomains
-        del cls.ad_treedomains
-    del cls.ad_domains
+    if cls.num_trusted_domains == 1:
```
Instead of having 2 separate code paths you can keep the old structure and add
```
    if cls.trusted_domains:
        del cls.trusted_master
        del cls.trusted_replicas
        del cls.trusted_clients
    del cls.trusted_domains
```

flo-renaud reviewed (commented):
Hi @amore17 
Thanks for the PR. Please find inline comments

flo-renaud reviewed (commented):
Hi @amore17 

the following property needs to be added to the Config class:
```
    @property
    def trusted_domains(self):
        return [d for d in self.domains if d.is_trusted_ipa_type]

```

flo-renaud commented on ipatests/pytest_ipa/integration/__init__.py:
```diff
@@ -413,6 +413,16 @@ def mh(request, class_integration_logs):
+                'trusted_master': 1,
+                'trusted_replica': cls.num_trusted_replicas,
+                'trusted_client': cls.num_trusted_clients,
+            }
```
The above lines should be
```
            {
                'master': 1,
                'replica': cls.num_trusted_replicas,
                'client': cls.num_trusted_clients,
            }
```

flo-renaud commented:
Hi @amore17 
LGTM. Let's first merge and deploy https://github.com/freeipa/freeipa-pr-ci/pull/506. When all the runners are updated, you will be able to submit this PR to the official https://github.com/freeipa/freeipa repo.

flo-renaud committed:
Custodia: in fips mode add -nomac or -nomacver to openssl pkcs12

In FIPS mode the command openssl pkcs12 fails unless the
export is called with -nomac and import with -nomacver

The command is used by custodia to export private keys from the
master and import them in the replica.

Fixes: https://pagure.io/freeipa/issue/9577

Signed-off-by: Florence Blanc-Renaud <flo@redhat.com>
Reviewed-By: Alexander Bokovoy <abokovoy@redhat.com>

stanislavlevin committed:
ipatests: make TestDuplicates teardowns order agnostic

Fixtures 'user4' and 'user5' track the same actual user 'tuser'.
If used together their teardowns can fail depending on the
order of execution.

With this change fixtures of TestDuplicates are simplified and
method-scoped.

Related: https://pagure.io/freeipa/issue/9571
Signed-off-by: Stanislav Levin <slev@altlinux.org>
Reviewed-By: Florence Blanc-Renaud <flo@redhat.com>

flo-renaud committed:
azure tests: move to fedora 40

Fedora 39 is approaching its EOL (2024-11-12),
move azure tests to fedora 40

Signed-off-by: Florence Blanc-Renaud <flo@redhat.com>
Reviewed-By: Alexander Bokovoy <abokovoy@redhat.com>

flo-renaud committed:
UnsafeIPAddress: pass flag=0 to IPNetwork

When parsing a string, the constructor tries to parse the value
as an IP Address first, or falls back to an IPNetwork with the
flags INET_PTON.

Use the flag 0 instead for an IPNetwork.

Fixes: https://pagure.io/freeipa/issue/9645
Signed-off-by: Florence Blanc-Renaud <flo@redhat.com>
Reviewed-By: Alexander Bokovoy <abokovoy@redhat.com>

flo-renaud committed:
azure pipeline: use latest version of DownloadPipelineArtifact task

update the task from v0 to v2

Signed-off-by: Florence Blanc-Renaud <flo@redhat.com>
Reviewed-By: Alexander Bokovoy <abokovoy@redhat.com>

yrro committed:
Fix a couple of instances of the "no-break control character" being used inadvertently

Fixes https://pagure.io/freeipa/issue/9665

Signed-off-by: Sam Morris <sam@robots.org.uk>
Reviewed-By: Alexander Bokovoy <abokovoy@redhat.com>

flo-renaud committed:
test_adtrust_install: add --use-krb5-ccache to smbclient command

With samba 4.20.4 the smbclient commands needs a ccache otherwise it
prompts for a password.

Fixes: https://pagure.io/freeipa/issue/9666
Signed-off-by: Florence Blanc-Renaud <flo@redhat.com>
Reviewed-By: Rob Crittenden <rcritten@redhat.com>
Reviewed-By: Alexander Bokovoy <abokovoy@redhat.com>

flo-renaud committed:
ipatests: provide a ccache to rpcclient deletetrustdom

With samba update to samba-4.20.4, rpcclient now needs a
ccache otherwise it prompts for a password.

Fixes: https://pagure.io/freeipa/issue/9667

Signed-off-by: Florence Blanc-Renaud <flo@redhat.com>
Reviewed-By: Rob Crittenden <rcritten@redhat.com>
Reviewed-By: Alexander Bokovoy <abokovoy@redhat.com>

flo-renaud committed:
test_ipahealthcheck: skip connectivity_and_data check

PKI removed the clones.check connectivity_and_data check in
11.5 and master branches. Skip the test depending on PKI version.
The most recent version on 11.5 is 11.5.4 and still contains the check,
hence skipping if version >= 11.5.5.

Fixes: https://pagure.io/freeipa/issue/9668

Signed-off-by: Florence Blanc-Renaud <flo@redhat.com>
Reviewed-By: Rob Crittenden <rcritten@redhat.com>

menonsudhir committed:
ipatests: Activate ssh in sssd.conf

This testcase checks that services: ssh
is included in the sssd.conf file when
ipa-client-install is successful.

Ref: https://pagure.io/freeipa/issue/9649

Signed-off-by: Sudhir Menon <sumenon@redhat.com>
Reviewed-By: Rob Crittenden <rcritten@redhat.com>

menonsudhir committed:
ipatests: Fixes for ipa-idrange-fix testsuite

This patch adds the line tasks.install_master(cls.master).
The kinit admin command fails with the below error as the
IPA is not configured on the test system

'ipa: ERROR: stderr: kinit: Configuration file does not specify default
realm when parsing name admin'

Signed-off-by: Sudhir Menon <sumenon@redhat.com>
Reviewed-By: Rob Crittenden <rcritten@redhat.com>

simmon-nplob committed:
Translated using Weblate (Korean)

Currently translated at 19.0% (930 of 4881 strings)

Translated using Weblate (Korean)

Currently translated at 19.0% (930 of 4881 strings)

Co-authored-by: 김인수 <simmon@nplob.com>
Translate-URL: https://translate.fedoraproject.org/projects/freeipa/master/ko/
Translation: freeipa/master
Reviewed-By: Florence Blanc-Renaud <flo@redhat.com>

rcritten committed:
Don't rely on removing the CA to uninstall the ACME depoyment

There has always been a pki-server commnd acme-remove. We were
not aware that it should be called prior to removing a CA. In
11.5.0 this is strongly encouraged by the PKI team. In 11.6.0
ACME is treated as a full subsystem so will be removed in the
future using pkidestroy -s ACME

The new class acmeinstance.ACMEInstance is introduced so its
uninstallation can be handled in a similar way as the other
PKI services via DogtagInstance. It is, right now, a pretty
thin wrapper.

We can discuss moving the ACME installation routines here at
some point. It would be ok as long as we don't have to introduce
another PKI restart as part of it.

In PKI 11.6.0 pkidestroy has new options to ensure a clean
uninstall: --remove-conf --remove-logs. Pass those options
into pkidestroy calls for 11.6.0+.

Clean up an additional IPA-generated file that needs to be
cleaned up during uninstall: /root/kracert.p12. 11.6.0 is
more sensitive to leftover files than previous versions.

Fixes: https://pagure.io/freeipa/issue/9673
Fixes: https://pagure.io/freeipa/issue/9674

Signed-off-by: Rob Crittenden <rcritten@redhat.com>
Reviewed-By: Alexander Bokovoy <abokovoy@redhat.com>

abbra committed:
selinux: allow Cockpit to use HTTP keytab on IPA servers

Cockpit can use GSSAPI authentication and has pretty good definition of
how to enable it: https://cockpit-project.org/guide/latest/sso.html.
These instructions work on IPA clients but they cannot be used on IPA
servers because IPA framework already owns HTTP/.. Kerberos service and
its keytab.

Luckily, there are two changes that need to be done to enable Cockpit
single sign-on with GSSAPI on IPA servers:

  - create a symlink /etc/cockpit/krb5.keytab to
    /var/lib/ipa/gssproxy/http.keytab

  - add SELinux policy to allow cockpit_session_t to operate on
    /var/lib/ipa/gssproxy/http.keytab file

For existing installation an upgrade process would restore SELinux
context of the http.keytab file to the new value.

Note that Cockpit documentation above also talks about Kerberos service
modifications to enable delegation.  These modifications should not be
done for IPA servers' HTTP services, as these services are already
enabled to handle delegation.

Fixes: https://pagure.io/freeipa/issue/9675

Signed-off-by: Alexander Bokovoy <abokovoy@redhat.com>
Reviewed-By: Rob Crittenden <rcritten@redhat.com>

mreynolds389 committed:
Do not let user with an expired OTP token to log in if only OTP is allowed

If only OTP authentication is allowed, and a user tries to login with an
expired token, do not let them log in with their password. Forcing the
admin to intervene. If the user does not have an OTP token then allow
them to log in with a password until an OTP token is configured

Fixes: https://pagure.io/freeipa/issue/9387

Signed-off-by: Mark Reynolds <mreynolds@redhat.com>
Reviewed-By: Rob Crittenden <rcritten@redhat.com>
Reviewed-By: Alexander Bokovoy <abokovoy@redhat.com>
Reviewed-By: Julien Rische <jrische@redhat.com>

Commit:
Replace instances of del os.environ with os.environ.pop

Fixes: https://pagure.io/freeipa/issue/9450

Signed-off-by: Ian Brown <ian.brown@templeandwebster.com.au>
Signed-off-by: Ian Brown <ianbrown78@gmail.com>
Reviewed-By: Alexander Bokovoy <abokovoy@redhat.com>

yselkowitz committed:
spec: Use nodejs22 on RHEL 10 and ELN

nodejs22 is now the default nodejs version in RHEL 10 as well as ELN.

Signed-off-by: Yaakov Selkowitz <yselkowi@redhat.com>
Reviewed-By: Alexander Bokovoy <abbra@users.noreply.github.com>
Reviewed-By: Florence Blanc-Renaud <flo@redhat.com>

abbra committed:
Minimal test for Cockpit integration on IPA master

Add a test to share HTTP service keytab on IPA master between IPA and
Cockpit. The test configures Cockpit with IPA CA-issued certificate and
allows Cockpit to access IPA HTTP service keytab for authentication.

The test then attempts to authenticate with GSSAPI as admin user. A
successful result is when we receive CSRF token from the Cockpit as
the result of this authentication. This means we have logged in
successfully with Kerberos.

Fixes: https://pagure.io/freeipa/issue/9675

Signed-off-by: Alexander Bokovoy <abokovoy@redhat.com>
Reviewed-By: Rob Crittenden <rcritten@redhat.com>

abbra committed:
Extend nightly tests with Cockpit test

Fixes: https://pagure.io/freeipa/issue/9675

Signed-off-by: Alexander Bokovoy <abokovoy@redhat.com>
Reviewed-By: Rob Crittenden <rcritten@redhat.com>

flo-renaud committed:
Nightly test definition: use master_1repl topology for idrange_fix

The test test_ipa_idrange_fix is installing IPA server as it sets
topology=line. Its test definition should not use a template
that pre-installs the IPA server (ipaserver preinstalls IPA server
but master_1repl does not).

Signed-off-by: Florence Blanc-Renaud <flo@redhat.com>
Reviewed-By: Sudhir Menon <sumenon@redhat.com>
